api/buckets/service.go

```diff
@@ -398,13 +402,18 @@ func (s *Service) PushPath(server pb.API_PushPathServer) error {
 	}
 
 	buckPath := path.New(buck.Path)
-	dirpth, err := s.IPFSClient.Object().
-		AddLink(server.Context(), buckPath, filePath, pth, options.Object.Create(true))
+	dirpth, err := s.IPFSClient.Object().AddLink(server.Context(), buckPath, filePath, pth, options.Object.Create(true))
 	if err != nil {
 		return err
 	}
 	if err = s.IPFSClient.Pin().Update(server.Context(), buckPath, dirpth); err != nil {
-		return err
+		if err.Error() == "'from' cid was not recursively pinned already" {
+			if err = s.IPFSClient.Pin().Add(server.Context(), dirpth); err != nil {
+				return err
+			}
```

Comment:
Uhm, I see. This answers my last question. A bit worried about Error string changes and breaking this logic.
Maybe your wording about why the bucket now isn't pinned from start will light this.

Reply:
Yeah, there are not sentinels from the HTTP client :( Maybe an issue over there would be a good idea.

Reply:
Catching this error here just in case something weird happened with the bucket (as reported by some users... though that was an old report). Confusingly, this is not related to the logic in `createBucket`... but pinning the wrong version of the cid would also pop this error.

Reply:
Issue: https://github.com/ipfs/go-ipfs-http-client/issues/100

Reply:
Yep, sounds good.